### PR TITLE
Various tidy ups.

### PR DESCRIPTION
**Pull request by vext01**
A miscellaneous bag of minor annoyances fixed. See commit messages.

Companion PR ~~coming~~: https://github.com/softdevteam/yk/pull/90

I also considered deleting software tracing, as it's dead and bit-rotted. Opted not to just yet.

### CODE REVIEW

File: src/librustc_middle/sir.rs [hunk 1]
```diff
@@ -114,7 +114,6 @@ impl SirFuncCx<'tcx> {
                 symbol_name,
                 blocks,
                 flags,
-                num_locals,
```
Comment (ptersilie) [inline src/librustc_middle/sir.rs:117]: Should we also get rid of the comment on line 107 and fold 108/109 together?

**Comment by ptersilie** [comment]
One comment.

File: src/librustc_middle/sir.rs (hunk 1 above)
Comment (vext01) [reply src/librustc_middle/sir.rs:117]: Oops! yes!

**Comment by vext01** [comment]
Fixed. If that looks good, I'll do priming.

File: src/librustc_middle/sir.rs [hunk 2]
```diff
@@ -104,11 +104,9 @@ impl SirFuncCx<'tcx> {
             .yk_trace_inputs()
             .expect("couldn't find trace inputs lang item");
 
-        // FIXME get rid of the num_locals field.
-        let num_locals = mir.local_decls.len();
-        let local_decls = Vec::with_capacity(num_locals);
-
+        let local_decls = Vec::with_capacity( mir.local_decls.len());
```
Comment (ptersilie) [inline src/librustc_middle/sir.rs]: There's a superfluous space just before `mir`.

**Comment by ptersilie** [comment]
One minor comment. But you can prime after that!

**Comment by vext01** [comment]
Priming...

bors try

**Comment by vext01** [comment]
priming reverted, ready for bors.

**Comment by vext01** [comment]
Something odd happened here. Perhaps an out of date lock file. Investigating...

**Comment by vext01** [comment]
bors try

**Comment by vext01** [comment]
OK to squash that in?

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat.

**Pull request merged**